Heimdall/NSURLRequestExtensions.swift

```diff
@@ -76,18 +76,41 @@ public extension NSMutableURLRequest {
     /// :param: parameters The parameters to be encoded or `nil`.
     ///
     /// TODO: Tests crash without named parameter.
-    public func setHTTPBody(#parameters: [String: String]?) {
+    public func setHTTPBody(#parameters: [String: AnyObject]?) {
         if let parameters = parameters {
-            var parts = [String]()
-            for (name, value) in parameters {
-                let encodedName = name.stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)
-                let encodedValue = value.stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)
-                parts.append("\(encodedName!)=\(encodedValue!)")
+            var components: [(String, String)] = []
+            for (key, value) in parameters {
+                components += self.queryComponents(key, value)
             }
-
-            HTTPBody = "&".join(parts).dataUsingEncoding(NSUTF8StringEncoding)
+            var bodyString = join("&", components.map{"\($0)=\($1)"} as [String])
+            HTTPBody = bodyString.dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: false)
         } else {
             HTTPBody = nil
         }
     }
+    
+    //Taken from https://github.com/Alamofire/Alamofire/blob/master/Source/ParameterEncoding.swift#L136
+    func queryComponents(key: String, _ value: AnyObject) -> [(String, String)] {
```

Comment:
Is this supposed to be `internal` or should it be `private`? Whitespace in the comment.